### ISSUE TITLE
Implement SLURM Client

### ISSUE THREAD

**Issue by bschroeter** [ACCESS-NRI/hpcpy#28]
Following #1, and recent discussions, we should start thinking about implementing a SLURM client that is at least equivalent to the PBS implementation.

See https://slurm.schedmd.com/rosetta.pdf for a quick translation from one to the other.

Target v1

Ping @aidanheerdegen

**Comment by bschroeter** [comment]
I want to use this, but there is just not enough documentation to know if it does what I think it does... https://github.com/PySlurm/pyslurm

**Comment by aidanheerdegen** [comment]
It's a bit buried, but the API reference seems quite extensive

https://pyslurm.github.io/23.2/reference/job/

**Comment by bschroeter** [comment]
Yes, just looking at the man page for SLURM there are a bajillion flags and options. I am tempted to create a generic interface to maintain the agnostic flavour of HPCpy that perhaps uses this as a backend for users who want to go down that rabbit hole.

Also coming around to the idea of a `Job` class, to access all of those magical backend-y things.

**Comment by bschroeter** [comment]
Got access to Pawsey to test SLURM. Will update when there is something to report
